contracts/README.md

```diff
@@ -0,0 +1,180 @@
+# Devnet contracts for development
+
+This folder contains Cairo and Solidity contracts for developement of Devnet.
+If you wish you to check specifically one of the two chains contracts, please check:
+1. `solidity` folder for Ethereum related contracts.
+2. `cairo` folder for Starknet related contracts, and example of how to work with starknet without
+   running a L1 node.
+
+## Full example of e2e testing with Anvil and Devnet
+
+### Setup of the nodes
+Please ensure that you've [anvil](https://book.getfoundry.sh/getting-started/installation) installed (or you can do the same with HardHat, but the commands here are done with anvil).
+```bash
+anvil
+```
+
+For Starknet, ensure you've Devnet compiled and running with the following params:
+```bash
+cargo run -- --seed 42
+```
+
+Now both nodes are running, Devnet for Starkne and Anvil for Ethereum.
+
+Then, **open two terminals**:
+* one for Starknet then:
+  ```bash
+  cd contracts/cairo
+  scarb build
+  ```
+
+* one for Ethereum then:
+  ```bash
+  cd contracts/solidity
+  # on first run you may need this
+  forge install
+  forge build
+  ```
+
+### Etherum setup
+1. Use Devnet postman endpoint to load the `MockStarknetMessaging` contract:
+```bash
+curl -H 'Content-Type: application/json' \
+     -d '{"networkUrl": "http://127.0.0.1:8545"}' \
+     http://127.0.0.1:5050/postman/load_l1_messaging_contract
+```
+```json
+{
+    "messageContractAddress":"0x5fbdb2315678afecb367f032d93f642f64180aa3"
+}
+```
+
+2. Deploy the `L1L2.sol` contract in order to receive/send messages from/to L2.
+```bash
+# Ethereum terminal (contracts/solidity)
+source .env
+forge script script/L1L2.s.sol:Deploy --broadcast --rpc-url $ETH_RPC_URL --silent
+```
+```bash
+✅  [Success]Hash: 0x942cfaadc557f360b91e2bfe98e8246d87b8efb4bfe6c1803162cd4aa7a71e1d
+Contract Address: 0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512
+Block: 2
+Paid: 0.0013459867197597 ETH (346581 gas * 3.8836137 gwei)
+```
+
+3. Check balance is 0 for user `1`:
+```bash
+cast call 0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512 "get_balance(uint256)(uint256)" 0x1
+```
+```bash
+0
+```
+
+### Starknet contracts and send message to L1
+1. On the Starknet terminal, set [starkli](https://book.starkli.rs/installation) variables:
+```bash
+# Starknet terminal (contracts/cairo)
+source .env
+
+# Declare
+starkli declare target/dev/cairo_l1_l2.contract_class.json
```

Comment:
we can probably pre-deploy this contract on the devnet startup, thus saving these steps!

Reply:
I would also tend to follow @FabijanC point of view. The flow already includes the generation of the file.